internal/pkg/keeper/mock_keeper.go

```diff
@@ -0,0 +1,169 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package keeper
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type MockKeeper struct {
+	serviceStore map[string]RegistrationDTO
+	serviceLock  sync.Mutex
+}
+
+func NewMockKeeper() *MockKeeper {
+	mock := MockKeeper{
+		serviceStore: make(map[string]RegistrationDTO),
+	}
+
+	return &mock
+}
+
+func (mock *MockKeeper) Start() *httptest.Server {
+	testMockServer := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		if strings.HasSuffix(request.URL.Path, ApiRegisterRoute) {
+			switch request.Method {
+			case http.MethodPost:
+				mock.serviceLock.Lock()
+				defer mock.serviceLock.Unlock()
+
+				bodyBytes, err := io.ReadAll(request.Body)
+				if err != nil {
+					log.Printf("error reading request body: %s", err.Error())
+				}
+
+				var req AddRegistrationRequest
+				err = json.Unmarshal(bodyBytes, &req)
+				if err != nil {
+					log.Printf("error decoding request body: %s", err.Error())
+				}
+
+				resp, err := http.Get(req.Registration.HealthCheck.Type + "://" + req.Registration.Host + ":" + strconv.Itoa(req.Registration.Port) + req.Registration.HealthCheck.Path)
```

Comment:
Why is the mock making a HTTP request??? It should be mock processing the incoming HTTP request.

Reply:
Since we have implemented a private function [getRegistryByService](https://github.com/edgexfoundry/go-mod-registry/pull/138/files#diff-1284d0c429d2b38d42b7baa22190bec31c0e657e9132f5167f47b8a038a0f6a2R281) to check if the service registry exists, which can't be mocked. 
Thus we have set up a test server to simulate the service for the health check callback in [unit test](https://github.com/edgexfoundry/go-mod-registry/pull/138/files#diff-20005dee2615486653bcefcaae8a860bdfbdaba000cac360b1de9b24a7cc76b0R74-R88).

Reply:
@jackchenjc , I reread the code and now under stand that you are mocking the HealthCheck callback. I am good with this.